dali/kernels/imgproc/color_manipulation/brightness_contrast.h

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_H_
+
+#include "dali/util/ocv.h"
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+#include "dali/pipeline/data/types.h"
+
+namespace dali {
+namespace kernels {
+
+namespace brightness_contrast {
+
+
+/**
+ * Assumes HWC layout
+ */
+template<size_t ndims, class CoordinateType>
+TensorShape<ndims + 1> roi_shape(Box<ndims, CoordinateType> roi, size_t nchannels) {
+  assert(all_coords(roi.hi >= roi.lo) && "Cannot create a tensor shape from an invalid Box");
+  TensorShape<ndims + 1> ret;
+  auto e = roi.extent();
+  auto ridx = ndims;
+  ret[ridx--] = nchannels;
+  for (size_t idx = 0; idx < ndims; idx++) {
+    ret[ridx--] = e[idx];
+  }
+  return ret;
+}
+
+}  // namespace brightness_contrast
+
+
+template<typename InputType, typename OutputType, size_t ndims = 3>
+class BrightnessContrastCPU {
+ private:
+  static constexpr size_t spatial_dims = ndims - 1;
+
+ public:
+  using Roi = Box<spatial_dims, int>;
+
+
+  KernelRequirements
+  Setup(KernelContext &context, const InTensorCPU<InputType, ndims> &in, float brightness,
+        float contrast, const Roi *roi = nullptr) {
+    DALI_ENFORCE(!roi || all_coords(roi->hi >= roi->lo), "Region of interest is invalid");
+    auto adjusted_roi = AdjustRoi(roi, in.shape);
+    KernelRequirements req;
+    req.output_shapes = {TensorListShape<DynamicDimensions>(
```

Comment:
1. Weird formatting.
2. There's a bug in number of channels - hardcoded as 3, but should be `in.shape[channel_dim]` (consider adding `static constexpr size_t channel_dim = ndims - 1;`
3. The code below is arguably more readable (and possibly more efficient):
```
TensorListShape<ndims> out_shape = { roi_shape(adjusted_roi, in.shape[channel_dim]) };
req.output_shapes = { std::move(out_shape) };
```